doc: Use multipath descriptors in descriptors.md and linked test

anuragchvn-blip opened the pull request:
Updates documentation and `wallet_miniscript_decaying_multisig_descriptor_psbt.py` to use single multipath descriptors with `<0;1>` syntax instead of separate external/internal descriptors.

## Changes
- **doc/descriptors.md**: Update examples (lines 70-71) to use `/<0;1>/*` multipath syntax
- **doc/descriptors.md**: Update Basic Multisig Example instructions (line 179) to use single multipath descriptor
- **test/functional/wallet_miniscript_decaying_multisig_descriptor_psbt.py**: Refactor to use single multipath descriptor pattern matching `wallet_multisig_descriptor_psbt.py`

## Implementation
- `_get_xpub()` now extracts external descriptor and converts to multipath format
- `create_multisig()` imports single descriptor that expands to receive and change addresses
- Removed fake checksums from documentation examples
- Added clear comments documenting multipath convention

Fixes #34086

rkrux reviewed (commented):
Concept ACK 912d837326145c304662ca29a01ccd1d240946e8

Multi-path convention is indeed convenient and preferred.

rkrux commented:
No, it still fails with the same error. It's suggested and preferred that the PR authors run the tests in their system as well.
Ref: https://github.com/bitcoin/bitcoin/blob/516ae5ede44a8a6abc59e90f9d89913e254524f4/CONTRIBUTING.md?plain=1#L304-L314

anuragchvn-blip commented:
Fixed the test by matching the pattern from `wallet_multisig_descriptor_psbt.py`:

**Before:**
```python
desc = multisig.getdescriptorinfo(...)
"desc": desc["descriptor"],
"range": 1000,
```

**After:**
```python
multisig_desc = f"wsh(thresh(...))"
checksum = multisig.getdescriptorinfo(multisig_desc)["checksum"]
"desc": f"{multisig_desc}#{checksum}",
```

Removed the explicit `"range"` parameter. Bitcoin Core automatically expands `<0;1>/*` multipath descriptors into separate external (0) and internal/change (1) descriptors when imported this way. The previous approach prevented this expansion, causing the "No bech32 addresses available" error.

anuragchvn-blip committed:
doc: Use multipath descriptors in descriptors.md and linked test

Updates documentation and wallet_miniscript_decaying_multisig_descriptor_psbt.py
to use single multipath descriptors with <0;1> syntax instead of separate
external/internal descriptors.

Changes:
- doc/descriptors.md: Update examples to use /<0;1>/* multipath syntax with /0 and /1 notation
- doc/descriptors.md: Update Basic Multisig Example instructions
- test: Refactor to use single multipath descriptor pattern, matching wallet_multisig_descriptor_psbt.py

Implementation:
- _get_xpub() extracts external descriptor and converts to multipath format
- create_multisig() builds descriptor string, gets checksum, imports descriptor#checksum
- Multipath descriptor properly expands to external and internal/change descriptors

Fixes #34086

rkrux commented on doc/descriptors.md:
```diff
@@ -67,8 +67,8 @@ Output descriptors currently support:
 - `tr(c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5,sortedmulti_a(2,2f8bde4d1a07209355b4a7250a5c5128e88b84bddc619ab7cba8d569b240efe4,5cbdf0646e5db4eaa398f365f2ea7a0e3d419b7e0330e39ce92bddedcac4f9bc))` describes a P2TR output with the `c6...` x-only pubkey as internal key, and a single `multi_a` script that needs 2 signatures with 2 specified x-only keys, which will be sorted le ...
-- `wsh(sortedmulti(2,[6f53d49c/44h/1h/0h]tpubDDjsCRDQ9YzyaAq9rspCfq8RZFrWoBpYnLxK6sS2hS2yukqSczgcYiur8Scx4Hd5AZatxTuzMtJQJhchufv1FRFanLqUP7JHwusSSpfcEp2/0/*,[e6807791/44h/1h/0h]tpubDDAfvogaaAxaFJ6c15ht7Tq6ZmiqFYfrSmZsHu7tHXBgnjMZSHAeHSwhvjARNA6Qybon4ksPksjRbPDVp7yXA1KjTjSd5x18KHqbppnXP1s/0/*,[367c9cfa/44h/1h/0h]tpubDDtPnSgWYk8dDnaDwnof4ehcnjuL5VoUt1eW2MoAed1grPHuXPDnkX1fWMvXfcz3NqFxPbhqNZ3QBdYjLz ...
-- `wsh(thresh(4,pk([7258e4f9/44h/1h/0h]tpubDCZrkQoEU3845aFKUu9VQBYWZtrTwxMzcxnBwKFCYXHD6gEXvtFcxddCCLFsEwmxQaG15izcHxj48SXg1QS5FQGMBx5Ak6deXKPAL7wauBU/0/*),s:pk([c80b1469/44h/1h/0h]tpubDD3UwwHoNUF4F3Vi5PiUVTc3ji1uThuRfFyBexTSHoAcHuWW2z8qEE2YujegcLtgthr3wMp3ZauvNG9eT9xfJyxXCfNty8h6rDBYU8UU1qq/0/*),s:pk([4e5024fe/44h/1h/0h]tpubDDLrpPymPLSCJyCMLQdmcWxrAWwsqqssm5NdxT2WSdEBPSXNXxwbeKtsHAyXPpLkhUyKovtZ ...
+- `wsh(sortedmulti(2,[6f53d49c/44h/1h/0h]tpubDDjsCRDQ9YzyaAq9rspCfq8RZFrWoBpYnLxK6sS2hS2yukqSczgcYiur8Scx4Hd5AZatxTuzMtJQJhchufv1FRFanLqUP7JHwusSSpfcEp2/<0;1>/*,[e6807791/44h/1h/0h]tpubDDAfvogaaAxaFJ6c15ht7Tq6ZmiqFYfrSmZsHu7tHXBgnjMZSHAeHSwhvjARNA6Qybon4ksPksjRbPDVp7yXA1KjTjSd5x18KHqbppnXP1s/<0;1>/*,[367c9cfa/44h/1h/0h]tpubDDtPnSgWYk8dDnaDwnof4ehcnjuL5VoUt1eW2MoAed1grPHuXPDnkX1fWMvXfcz3NqFxPbhqNZ ...
```
Super nit if you have to retouch for some other reason:

To make it congruent with the similar line down below in this file.
```diff
- both receiving (/0) and change (/1) address derivation paths.
+ both receiving (`/0`) and change (`/1`) address derivation paths.
```

anuragchvn-blip replied:
Thanks for confirming. I’m happy to wait for the remaining ACKs. Please let me know if anything else is needed from my side. @rkrux

anuragchvn-blip replied:
@rkrux Its been a while no reviewers are up for reviewing.
Please do help me with that

rkrux reviewed (approved):
lgtm tACK 552bc82

I see in the previous commit https://github.com/bitcoin/bitcoin/commit/912d837326145c304662ca29a01ccd1d240946e8 that failed the test, only the "receive" descriptor was being imported without the "change" descriptor because the "descriptor" property in the response of `importdescriptors` RPC returns only the first expanded descriptor in case of _multipath_. 

Thus, the "Transaction needs a change address, but we can't generate it." error while creating the transaction seems correct. The latest commit 552bc82b17961b86ae1964e817ba89ee7bfd985f correctly fixes the issue by importing the complete _multipath_ descriptor.

achow101 commented:
ACK 552bc82b17961b86ae1964e817ba89ee7bfd985f
